src/pages/ProductInner/ProductInner.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+
+import { Preloader } from '../../components/Preloader/Preloader';
+import { ErrorMessage } from '../../components/ErrorMesage/ErrorMessage';
+
+import './ProductInner.scss';
+
+export function ProductInner() {
+  const { id } = useParams();
+  const [product, setPrdouct] = useState(null);
+  const [productImageUrl, setPdouctImageUrl] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    fetch(`https://api.thedogapi.com/v1/breeds/${id}`)
+      .then((res) => {
+        setLoading(false);
+        if (!res.ok) throw new Error();
+        return res.json();
+      })
+      .then((data) => setPrdouct(data))
+      .catch(() => setIsError(true));
+  }, []);
+
+  useEffect(() => {
+    product &&
+      fetch(`https://api.thedogapi.com/v1/images/${product.reference_image_id}`)
+        .then((res) => {
+          if (!res.ok) throw new Error();
+          return res.json();
+        })
+        .then((data) => setPdouctImageUrl(data.url))
+        .catch(() => setIsError(true));
+  }, [product]);
+
+  return (
+    <div className="ProductInner">
+      {loading && <Preloader />}
+      {isError && <ErrorMessage />}
+      {productImageUrl && (
+        <div className="container">
+          <Link className="ProductInner__link" to="/products">
+            Go to dogs list
+          </Link>
+          <div className="ProductInner__content">
+            <div className="ProductInner__image-container">
+              <img
+                className="ProductInner__image"
+                src={productImageUrl}
+                alt="fine_dog"
+              />
+            </div>
+            <div className="ProductInner__info-container">
+              <h2 className="ProductInner__title">{product.name}</h2>
```

Comment:
I would add question mark before accessing properties of the product like this `product?.name` in order to get the data safely

The same for all the properties below

or, alternatively, wrap all the product details information to the `product && (...some makup)` condition